### PR TITLE
Music auth

### PR DESCRIPTION
**Pull request by edadesd**
The music module was failing because all requests now need to be authenticated, as mentioned in #224 . I made the module authenticate via the Client Credentials flow, because we don't need to authorize on behalf of the user.

To accommodate this patch, @swapagarwal will need to register the app with Spotify at: https://developer.spotify.com/my-applications/

Then, generate the key for config.py by base64-encoding the client_id and client secret together in the following format, without the brackets:

[client_id]:[client secret]

I was able to get this to work using the encoder at https://www.base64encode.org/ . This PR also adds a line to config.py for the Spotify key.

### CODE REVIEW

**Comment by edadesd** [comment]
Made the changes requested by Stickler.

**Comment by edadesd** [comment]
Squashed commits to clean up history.

File: modules/src/music.py [hunk 1]
```diff
@@ -10,9 +12,19 @@
...
+
+        r = requests.post(auth_url, headers=headers, data=payload)
+
+        key = r.json()['access_token']
+
         music = entities['music'][0]['value']
         with requests_cache.enabled('music_cache', backend='sqlite', expire_after=3600):
-            r = requests.get('https://api.spotify.com/v1/search?q=' + music + '&type=track')
+            headers = {'Authorization': 'Bearer ' + key}
+            r = requests.get('https://api.spotify.com/v1/search?q=' + music + '&type=track&limit=1', headers=headers)
```
Comment (swapagarwal) [inline modules/src/music.py]: Why limit = 1?

File: modules/src/music.py [hunk 2]
```diff
@@ -10,9 +12,19 @@
...
     output = {}
     try:
+
+        auth_url = "https://accounts.spotify.com/api/token"
+        headers = {'Authorization': 'Basic ' + config.SPOTIFY_API_KEY}
+        payload = {'grant_type': 'client_credentials'}
+
+        r = requests.post(auth_url, headers=headers, data=payload)
+
+        key = r.json()['access_token']
```
Comment (swapagarwal) [inline modules/src/music.py]: Hmm, this token would be fetched everytime? Let's think of avoiding this somehow.

**Comment by swapagarwal** [comment]
Wow, you really have provided detailed instructions. Thanks Daniel!

File: modules/src/music.py (hunk 1 above)
Comment (edadesd) [inline modules/src/music.py]: Based on #208 I assumed we only used the first result. But looking at the rest of the module, I guess not.

File: modules/src/music.py (hunk 2 above)
Comment (edadesd) [inline modules/src/music.py]: Maybe save the token in a file, attempt to re-use it. If that fails, repeat the query and request a new token.

**Comment by edadesd** [comment]
Pushed a commit to remove the limit to one response.

File: modules/src/music.py (hunk 2 above)
Comment (edadesd) [inline modules/src/music.py]: Better yet, we can save it in config.py to avoid needing to create any new files.

**Comment by edadesd** [comment]
I made it store the access token in a file (music_token.json) and attempt to use the stored access token at the start of each script run. If the API returns 401 for unauthorized, it asks for a new token and stores that in the file, then repeats the request with the stored token. This allows the module to use a retrieved token until it expires.

**Comment by edadesd** [comment]
Rebased to current master.

**Comment by edadesd** [comment]
Rebased to current master.

**Comment by swapagarwal** [comment]
Thanks @edadesd! 👍 
I checked the module with the query "linkin park songs" and it's working as expected. 🎉

**Pull request merged**